src/boot/memory.c

```diff
@@ -367,6 +414,303 @@ void *load_to_fixed_pool_addr(u8 *destAddr, u8 *srcStart, u8 *srcEnd) {
     return dest;
 }
 
+#define PACKED __attribute__((packed))
+
+#define __GET_UNALIGNED_T(type, ptr) ({						\
+	const struct { type x; } PACKED *__pptr = (typeof(__pptr))(ptr);	\
+	__pptr->x;								\
+})
+
+#define __PUT_UNALIGNED_T(type, val, ptr) do {					\
+	struct { type x; } PACKED *__pptr = (typeof(__pptr))(ptr);		\
+	__pptr->x = (val);							\
+} while (0)
+
+#define GET_UNALIGNED8(ptr)	__GET_UNALIGNED_T(uint64_t, (ptr))
+#define PUT_UNALIGNED8(val, ptr) __PUT_UNALIGNED_T(uint64_t, (val), (ptr))
+#define GET_UNALIGNED4(ptr)	__GET_UNALIGNED_T(uint32_t, (ptr))
+#define GET_UNALIGNED4S(ptr)	__GET_UNALIGNED_T(int32_t, (ptr))
+#define PUT_UNALIGNED4(val, ptr) __PUT_UNALIGNED_T(uint32_t, (val), (ptr))
+
+#define EXPECT(expr,value)    (__builtin_expect ((expr),(value)) )
```

Comment:
remove likely/unlikely macros and their usages

Reply:
But first please confirm they actually affect compiled output; there could be value in adding them to macros.h.

Reply:
I decided to keep it for now, it is nicer for readability purposes.

Reply:
Removed likely/unlikely and optimized for size, reduce the size of the function by 30%. Worth it given it is only executed on emu anyways.